File: Backend/src/main/java/com/luckyseven/backend/domain/budget/dto/BudgetReadResponse.java
package com.luckyseven.backend.domain.budget.dto;

import com.luckyseven.backend.domain.budget.entity.CurrencyCode;
import jakarta.validation.constraints.NotNull;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import lombok.Builder;
import lombok.Getter;

public record BudgetReadResponse (
    @NotNull Long id,
    @NotNull LocalDateTime updatedAt,

    @NotNull Long setBy,
    @NotNull BigDecimal totalAmount,
    @NotNull BigDecimal balance,
    @NotNull CurrencyCode foreignCurrency,

    BigDecimal avgExchangeRate,
    BigDecimal foreignBalance
) { }

File: Backend/src/main/java/com/luckyseven/backend/domain/budget/dto/BudgetReadResponse.kt
package com.luckyseven.backend.domain.budget.dto

import com.luckyseven.backend.domain.budget.entity.CurrencyCode
import jakarta.validation.constraints.NotNull
import java.math.BigDecimal
import java.time.LocalDateTime

data class BudgetReadResponse(

    val id: Long,
    val updatedAt: LocalDateTime,

    val setBy: Long,
    val totalAmount: BigDecimal,

    val balance: BigDecimal,
    val foreignCurrency: CurrencyCode,

    val avgExchangeRate: BigDecimal?,
    val foreignBalance: BigDecimal?
)
